corehq/apps/export/views/download.py

```diff
@@ -437,22 +441,21 @@ def prepare_form_multimedia(request, domain):
         })
 
     export = view_helper.get_export(export_specs[0]['export_id'])
-    datespan = filter_form.cleaned_data['date_range']
-    user_types = filter_form.get_es_user_types(filter_form_data)
+    filters = filter_form.get_export_filters(request, filter_form_data)
+    export_es_query = get_export_query(export, filters)
 
-    if media_export_is_too_big(domain, export.app_id, export.xmlns, datespan, user_types):
+    if media_export_is_too_big(export_es_query):
         return json_response({
             'success': False,
             'error': _("This is too many files to export at once.  "
                        "Please modify your filters to select fewer forms."),
         })
 
     download = DownloadBase()
-    download.set_task(build_form_multimedia_zip.delay(
+    download.set_task(build_form_multimedia_zipfile.delay(
         domain=domain,
         export_id=export.get_id,
-        datespan=datespan,
-        user_types=user_types,
+        export_es_query=export_es_query,
```

Comment:
Is passing ES queries to tasks a thing we do frequently? It doesn't seem like a great idea to be passing an ES query object into a task since it will be serialized and stored in a queue for some time. I think this could make future ES query code changes more difficult because we will need to worry about maintaining compatibility with serialized queries, which seems hard to do properly and easy to forget.

Is it expensive to rebuild the ES query in the task? It seems safer to pass the data in `filters` or the data in `filter_form_data` and call `get_export_query` again inside the task.

Reply:
Interesting. That sounds better for sure.

If we want to build the filters in the task we would need to additionally pass the following to the task
1. filter_form_class which is `EmwfFilterFormExport`
2. filter_form_data which should be a simple JSON

Additionally, then in the task
1. we get the user's timezone
2. construct the form class object
3. fetch the export instance
4. and then get the filters.

Reply:
I see that for other exports we pass the ES query filters, https://github.com/dimagi/commcare-hq/blob/9c62b945fb3d03d9fcafec4b6f075a31e56e77f1/corehq/apps/export/views/download.py#L334-L342

Is that the better thing to do?

Reply:
I imagine that's better, but don't have any personal experience with it. @esoergel, do you have an opinion?

Reply:
cool, i will update to that anyway. This would bring all exports to have the same implementation.

Reply:
https://github.com/dimagi/commcare-hq/pull/29477/commits/6261b0d1250b90ac6b8f88a30fcc1e236a24b240

Reply:
It does seem like using whatever exports currently use would be the safest approach - that way if we do need to content with this problem, at least it'll be the same problem in both places.

Export filters are a little weird - they wrap the ESQuery filters in objects with only one method.

https://github.com/dimagi/commcare-hq/blob/d6f3f9a27cccf7903fc9b09522f0237c53f12554/corehq/apps/export/filters.py#L64-L70

I'm not sure why it's that way, seems like a lot of extra boilerplate.  At any rate, this means we pass around lists of those filter objects, which I'm guessing are pickled.  Does that mean if they're loaded into code on the other side, the delayed call to the underlying ESQuery filter would happen on the new code version?  That sounds advantageous.

Reply:
> It does seem like using whatever exports currently use would be the safest approach

Cool, that is what I did in the commit i linked in the thread above, so we should be good here.

> That sounds advantageous

Can you add what is the advantage? I am assuming it's not in this PR but someday is there are changes to how the ESQuery is made, it would not break the waiting export tasks (or any tasks that pass filters).